### PR TITLE
add handle keyboard focus for accessibility

### PR DESCRIPTION
**Pull request by vanpertsch**
add focus-trap (https://github.com/focus-trap/focus-trap) for accessibility

https://user-images.githubusercontent.com/38529028/165922863-7cb781d4-2b3c-4a93-9c5f-06d6b26ba6c6.mp4

Signed-off-by: Vanessa Pertsch <vanessa.pertsch@posteo.de>

### CODE REVIEW

File: src/components/Modal/Modal.vue [hunk 1]
```diff
@@ -525,6 +527,59 @@ export default {
...
+					firstFocusable.focus()
+				}
+			})
+		},
+		/**
+		 * set focus trap
+		 *
+		 * @param e
+		 */
+		checkForTabKeyEvent(e) {
```
Comment (PVince81) [inline src/components/Modal/Modal.vue:558]: did you know about the "inert" attribute ? https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert

it's not supported yet (or experimentally) by browsers but there's a polyfill for it

Comment (vanpertsch) [reply src/components/Modal/Modal.vue]: @PVince81 thanks. I thought maybe [https://github.com/focus-trap/focus-trap](https://github.com/focus-trap/focus-trap) is a good solution.

File: src/components/Modal/Modal.vue [hunk 2]
```diff
@@ -525,6 +527,32 @@ export default {
 				this.slideshowTimeout.clear()
 			}
 		},
+		/**
+		 * Get List of focusableelmets from content container
+		 */
+		getFocusableList() {
+			const contentContainer = this.$refs.mask.querySelector('.modal-container')
+			const focusableList = contentContainer.querySelectorAll(
```
Comment (PVince81) [inline src/components/Modal/Modal.vue:535]: I think now you don't need to query all, just the first element ?
basically rename the function to "getFirstFocusableElement" ? (assuming the focus trap doesn't do this already automatically)

Comment (vanpertsch) [reply src/components/Modal/Modal.vue:535]: ...i do not longer need  getFocusableList at all. forgot to remove it.

**Review by skjnldsv — APPROVED** [review]

File: src/components/Modal/Modal.vue [hunk 3]
```diff
@@ -525,6 +571,24 @@ export default {
 				this.slideshowTimeout.clear()
 			}
 		},
+		/**
+		 * Add focus trap for accessibility.
+		 */
+		useFocusTrap() {
+			const contentContainer = this.$refs.mask
+			this.$nextTick(function() {
```
Comment (artonge) [inline src/components/Modal/Modal.vue:579]: Do you know why it wasn't working without it? Could be nice to add a comment :)

File: src/components/Modal/Modal.vue [hunk 4]
```diff
@@ -525,6 +571,24 @@ export default {
...
+				this.focusTrap = createFocusTrap(contentContainer, {
+					onActivate: () => contentContainer.classList.add('is-active'),
+					onDeactivate: () => contentContainer.classList.remove('is-active'),
```
[the remark this answers is not in the document]
Comment (vanpertsch) [reply src/components/Modal/Modal.vue:582]: You are right, we don not need it.

File: src/components/Modal/Modal.vue (hunk 3 above)
Comment (vanpertsch) [reply src/components/Modal/Modal.vue:580]: added

**Review by artonge — APPROVED** [review]
:+1:

**Comment by szaimen** [comment]
Does it also work with multiple modals on top of each other e.g. like the text app uses?

**Comment by skjnldsv** [comment]
> Does it also work with multiple modals on top of each other e.g. like the text app uses?

Tested, works fine
```vue
<template>
	<div>
		<button @click="modal = true">Show Modal</button>
		<modal v-if="modal" @close="modal = false" size="small">
			<div class="modal__content">Hello world</div>

			<button @click="modal2 = true">Show Modal 2</button>
			<modal v-if="modal2" @close="modal2 = false" size="small">
				<div class="modal__content">Hello world 2</div>
			</modal>

		</modal>
	</div>
</template>
<style scoped>    
	.modal__content {
		margin: 50px;
		text-align: center;
	}

</style>
<script>
export default {
	data() {
		return {
			modal: false,
			modal2: false
		}
	},
}
</script>
</script>
```

**Pull request merged**